crt/glibc_stub.c

```diff
@@ -0,0 +1,224 @@
+#include <assert.h>
+#include <bits/alltypes.h>
+#include <resolv.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ucontext.h>
+
+#define weak_alias(old, new) \
+    extern __typeof(old) new __attribute__((__weak__, __alias__(#old)))
+
+__thread int errno;
+
+const char* const sys_errlist[] = {};
+const int sys_nerr = 0;
+
+void __makecontext(ucontext_t* ucp, void (*func)(void), int argc, ...)
+{
+    assert(0);
+}
+weak_alias(__makecontext, makecontext);
+
+int setcontext(const ucontext_t* ucp)
+{
+    assert(0);
+    return -1;
+}
+
+typedef struct __mbstate_t
+{
+    unsigned __opaque1, __opaque2;
+} mbstate_t;
+size_t __mbrlen(const char* s, size_t n, mbstate_t* ps)
+{
+    assert(0);
+    return 0;
```

Comment:
https://www.man7.org/linux/man-pages/man3/mbrlen.3.html - 0 return value indicates a null wide character was recognized. -1 return value if if an invalid multibyte sequence was encountered. -1 is a better error return to prevent the calling function from continue execution.